mbedtls/tests/client_server.rs

```diff
@@ -137,17 +139,20 @@ mod test {
             let server_max_ver = config[3];
             let exp_version = config[4];
 
+            if (client_max_ver < 3 || server_max_ver < 3) && !cfg!(feature="legacy_protocols") {
```

Comment:
It would be nice to test the negative cases, but I guess that would require support on either the client or the server so that doesn't work with conditional compilation.